[TTAHUB-2003] Modal button spacing

nvms opened the pull request:
## Description of change

Just adds some space between the buttons to more closely match USWDS modal button spacing.

## How to test


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2003


## Checklists

### Every PR

<!-- Add details to each completed item -->
- [x] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

thewatermethod reviewed (changes requested):
looks great, change also needs to be made all the places we use the "VanillaModal" component

nvms commented:
> looks great, change also needs to be made all the places we use the "VanillaModal" component

I think I found all these places and updated them. I also added a `.usa-button--subtle` variant to be used instead of an unstyled button inside of a modal. It's basically just a transparent `.usa-button`, so it has the right size, padding, etc. such that it looks (IMO) nice as a subtle modal button:

https://github.com/HHS/Head-Start-TTADP/assets/17074357/caf278c9-2fb3-4ead-b05f-fbc4a87829c8

thewatermethod commented on frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js:
```diff
@@ -193,7 +193,7 @@ const GoalPicker = ({
           Keep objective
         </Button>
-        <Button type="button" onClick={onRemove} unstyled>Remove objective</Button>
+        <Button type="button" onClick={onRemove} className="usa-button--unstyled">Remove objective</Button>
```
Should this use the subtle variant?

nvms replied:
Yep! 686bcfa536020805787e01397cde4eb1e1163d35

thewatermethod commented:
Looks good - ```main``` has the updated dependencies which will fix the CI failure
